src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -0,0 +1,265 @@
+                                        accessibilityRole="button"
+                                        onPress={() => {
+                                            togglePlayCurrentVideo();
+                                        }}
+                                        style={styles.flex1}
+                                    >
+                                        <Video
+                                            ref={videoPlayerRef}
+                                            style={[styles.w100, styles.h100, videoPlayerStyle]}
+                                            videoStyle={[styles.w100, styles.h100, videoStyle]}
+                                            source={{
+                                                uri: sourceURL,
+                                            }}
+                                            shouldPlay={false}
+                                            useNativeControls={false}
+                                            resizeMode={resizeMode}
+                                            isLooping={isLooping}
+                                            onReadyForDisplay={onVideoLoaded}
+                                            onPlaybackStatusUpdate={handlePlaybackStatusUpdate}
+                                            onFullscreenUpdate={handleFullscreenUpdate}
+                                        />
+                                    </PressableWithoutFeedback>
+                                </View>
+                            )}
+
+                            {(isLoading || isBuffering) && <FullScreenLoadingIndicator style={[styles.opacity1, styles.bgTransparent]} />}
+
+                            {shouldShowVideoControls && !isLoading && (isPopoverVisible || isHovered || canUseTouchScreen) && (
+                                <VideoPlayerControls
+                                    duration={duration}
+                                    position={position}
+                                    url={url}
+                                    videoPlayerRef={videoPlayerRef}
+                                    isPlaying={isPlaying}
+                                    small={shouldUseSmallVideoControls}
+                                    style={videoControlsStyle}
+                                    togglePlayCurrentVideo={togglePlayCurrentVideo}
+                                    showPopoverMenu={showPopoverMenu}
+                                />
+                            )}
+                        </View>
+                    )}
+                </Hoverable>
+            </View>
+            <VideoPopoverMenu
+                isPopoverVisible={isPopoverVisible}
+                hidePopover={hidePopoverMenu}
+                anchorPosition={popoverAnchorPosition}
+            />
+        </>
+    );
+}
+
+BaseVideoPlayer.propTypes = videoPlayerPropTypes;
+BaseVideoPlayer.defaultProps = videoPlayerDefaultProps;
```

Comment:
We changed the guidelines to use object destructuring to assign default values to optional component props and remove the usage of `defaultProps`. According to https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#destructuring
> Always use destructuring to get prop values. Destructuring is necessary to assign default values to props.

Reply:
This file should not be modified in this PR, I added it as an accident. It's already removed from here.